src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Helper/Base64ValidatorHelper.cs

```diff
@@ -9,23 +9,29 @@ internal static partial class Base64Helper
     {
         internal static bool IsValid<T, TBase64Validatable>(TBase64Validatable validatable, ReadOnlySpan<T> base64Text, out int decodedLength)
             where TBase64Validatable : IBase64Validatable<T>
+            where T : struct
         {
             int length = 0, paddingCount = 0;
+            T lastChar = default;
 
             if (!base64Text.IsEmpty)
             {
 #if NET
-                while (true)
+                while (!base64Text.IsEmpty)
                 {
-
                     int index = validatable.IndexOfAnyExcept(base64Text);
                     if ((uint)index >= (uint)base64Text.Length)
                     {
                         length += base64Text.Length;
+                        lastChar = base64Text[base64Text.Length - 1];
                         break;
                     }
 
                     length += index;
+                    if (index > 0)
```

Comment:
Is this really a check for index != 0? We just validated that it's not negative in the previous if check.

Reply:
Yes, that is true